### PR TITLE
FIX: update colspecs to handle numeric types

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Context: I am trying out this sqlalchemy dialect to connect DuckDB with ibis (https://github.com/ibis-project/ibis). So I might do some contributions for things I run into.

This is a first, as I ran into an error when doing a basic query:

```
            else:
                raise exc.InvalidRequestError(
>                   "Unknown PG numeric type: %d" % coltype
                )
E               TypeError: %d format: a number is required, not str

../../../miniconda3/envs/test-duckdb/lib/python3.9/site-packages/sqlalchemy/dialects/postgresql/psycopg2.py:525: TypeError
```

This was from the psycopg2 driver which adds a custom `_PGNumeric` type (https://github.com/sqlalchemy/sqlalchemy/blob/70a33dd32f2609f2de6421c3b28ef4cc4496fdae/lib/sqlalchemy/dialects/postgresql/psycopg2.py#L500). But that expects that the `col_type` in the cursor's description is a number, while duckdb just uses "NUMBER" (https://github.com/duckdb/duckdb/blob/a776543eec63d16134af7583122252537b89677f/tools/pythonpkg/src/pyresult.cpp#L326)

### CODE REVIEW

File: duckdb_engine/tests/test_pandas.py [hunk 1]
```diff
@@ -117,3 +119,18 @@ def run_query(query: str, chunksize: Optional[int]) -> None:
         assert len(chunks[0]) == chunksize
         # Assert that the expected number of chunks was returned.
         assert (sample_rowcount / chunksize) == len(chunks)
+
+
+def test_read_sql_duckdb_table(tmp_path):
```
Comment (jorisvandenbossche) [inline duckdb_engine/tests/test_pandas.py:123]: I added this test which is partly duplicative with the test above, but written in a different style: I am first creating a table with duckdb, and then reading that with pandas/sqlalchemy (and then `read_sql("table_name")` actually seems to work ..). 

I am also not sure why this way, this test was failing before my patch, while the existing tests were not failing ..

**Review by Mause — APPROVED** [review]

**Comment by Mause** [comment]
Hey, thanks for fixing this! Are you able to fix up the lint error and run the tests on your branch?

**Comment by jorisvandenbossche** [comment]
@Mause I fixed the mypy issue (at least locally pre-commit is passing) 
(you'll need to approve the workflow again for the tests to run here)

**Comment by Mause** [comment]
Turns out the triggers on the CI were setup incorrectly for pull requests from forks - fixed that now 😅

**Comment by Mause** [comment]
I've included this in 0.1.8rc2 💜

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request merged**